Documentation/Articles/querying_for_data.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Querying for Data
+description: About Reading data
+keywords: read model, query, ddd, read, cqrs,  
+author: pavsaund
+---
+
+
+# Querying for Data
+
+## Introduction
+The query or read side of the CQRS pattern is all about exposing the current state of a system in a format that is ready to be consumed by interaction layers (ie: web-based user interface). The state should be exposed through models that are optimized for querying, and is usually updated by changes made to a system through events.
+
+{{% notice note %}}
+Dolittle exposes read data through two different mechanisms:
+ 
+ - [Read Models](https://github.com/dolittle/Runtime/tree/master/Documentation/read/read_model.md)
+ - [Queries](https://github.com/dolittle/Runtime/tree/master/Documentation/read/query.md)
+{{% /notice %}}
+
+### Updating Read Data
+[Event Processors](LINK TO EVENT PROCESSORS) are responsible for consuming events in the system and updating read-models.
+
+### Storing Data 
+#### Optimize for querying
+Unlike traditional relational data models that are spread across tables, read data should be optimized as much as possible to the clients consuming that data. This means creating models that can be consumed directly without additional queries. One of the of the benefits is that reading data becomes lightweight, fast and allows for good user experiences.
+
+Duplicating data in 
```

Comment:
This just peters out - is something missing?

Something like "Duplicating data in several places to optimise for reads is considered a good thing when using Dolittle. It is the responsibility of the Command-side to keep these optimised structures up-to-date and correct. The Read-side merely gives the data it has access to with minimal logic and fuss. This might seem foreign and weird, but it simplifies both the read and write sides a lot, and lets systems be responsive and performant on the read-side independently of the write-side."

Reply:
oh, wow - I must have dozed off while writing this.